src/main/scala/com/databricks/labs/overwatch/pipeline/Snapshot.scala

```diff
@@ -0,0 +1,343 @@
+        }else{
+          println(s"First Time Write to  ${sourceName}")
+          logger.log(Level.INFO, s"Beginning write to ${sourceName}")
+          val msg = s"Checkpoint Path Set: ${checkPointLocation} - proceeding with streaming write"
+          logger.log(Level.INFO, msg)
+          val beginMsg = s"Stream to ${sourceName} beginning."
+          logger.log(Level.INFO, beginMsg)
+          var streamWriter = rawStreamingDF.writeStream.outputMode("append").format("delta").option("checkpointLocation", checkPointLocation)
+            .queryName(s"StreamTo_${sourceName}")
+
+          streamWriter = if (cloneSpec.mode == WriteMode.overwrite) { // set overwrite && set overwriteSchema == true
+            streamWriter.option("overwriteSchema", "true")
+          } else { // append AND merge schema
+            streamWriter
+              .option("mergeSchema", "true")
+          }
+          val streamWriter1 = streamWriter
+            .asInstanceOf[DataStreamWriter[Row]]
+            .option("path", targetLocation)
+            .start()
+
+          val streamManager = Helpers.getQueryListener(streamWriter1,workspace.getConfig, workspace.getConfig.auditLogConfig.azureAuditLogEventhubConfig.get.minEventsPerTrigger)
+          spark.streams.addListener(streamManager)
+          val listenerAddedMsg = s"Event Listener Added.\nStream: ${streamWriter1.name}\nID: ${streamWriter1.id}"
+          logger.log(Level.INFO, listenerAddedMsg)
+
+          streamWriter1.awaitTermination()
+          spark.streams.removeListener(streamManager)
+          logger.log (Level.INFO, s"Streaming COMPLETE: ${cloneSpec.source} --> ${cloneSpec.target}")
+          CloneReport(cloneSpec, s"Streaming For: ${cloneSpec.source} --> ${cloneSpec.target}", "SUCCESS")
+        }
+
+      } catch {
+        case e: Throwable if (e.getMessage.contains("is after the latest commit timestamp of")) => {
+          val failMsg = PipelineFunctions.appendStackStrace(e)
+          val msg = s"SUCCESS WITH WARNINGS: The timestamp provided, ${cloneSpec.asOfTS.get} " +
+            s"resulted in a temporally unsafe exception. Cloned the source without the as of timestamp arg. " +
+            s"\nDELTA ERROR MESSAGE: ${failMsg}"
+          logger.log(Level.WARN, msg)
+          CloneReport(cloneSpec, s"Streaming For: ${cloneSpec.source} --> ${cloneSpec.target}", msg)
+        }
+        case e: Throwable => {
+          val failMsg = PipelineFunctions.appendStackStrace(e)
+          CloneReport(cloneSpec, s"Streaming For: ${cloneSpec.source} --> ${cloneSpec.target}", failMsg)
+        }
+      }
+    }).toArray.toSeq
+
+    val cloneReportPath = s"${snapshotRootPath}/clone_report/"
+    cloneReport.toDS.write.mode("append").option("mergeSchema", "true").format("delta").save(cloneReportPath)
+  }
+
+
+  private[overwatch] def buildCloneSpecs(
+                                          cloneLevel: String,
+                                          sourceToSnap: Array[PipelineTable]
```

Comment:
sourceToSnap should be sourcesToSnap as it's an array

Reply:
Done